controllers/dnspolicies_validator.go

```diff
@@ -0,0 +1,54 @@
+package controllers
+
+import (
+	"context"
+	"sync"
+
+	"github.com/samber/lo"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/kuadrant/policy-machinery/controller"
+	"github.com/kuadrant/policy-machinery/machinery"
+
+	kuadrantv1alpha1 "github.com/kuadrant/kuadrant-operator/api/v1alpha1"
+	"github.com/kuadrant/kuadrant-operator/pkg/library/kuadrant"
+)
+
+func NewDNSPoliciesValidator() *DNSPoliciesValidator {
+	return &DNSPoliciesValidator{}
+}
+
+type DNSPoliciesValidator struct{}
+
+func (r *DNSPoliciesValidator) Subscription() controller.Subscription {
+	return controller.Subscription{
+		ReconcileFunc: r.validate,
+		Events: []controller.ResourceEventMatcher{
+			{Kind: &machinery.GatewayGroupKind},
+			{Kind: &kuadrantv1alpha1.DNSPolicyGroupKind},
+		},
```

Comment:
Do we also want events for DNSRecord? As DNSPolicy owns DNSRecord and if it is changed manually, we would want to set it back. Or is DNSPolicy enough to trigger that behaviour?

Reply:
This even this on the validation task of the policy so i don't believe this is required here. No change to the record will cause the policy to be invalidated.